module/VuFind/src/VuFind/Config/Feature/SecretTrait.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * Trait to import secret from file rather than a hardcoded config
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Michigan State University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Robby ROUDON <roudonro@msu.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Config\Feature;
+
+use Laminas\Config\Config;
+
+/**
+ * Trait to import secret from file rather than a hardcoded config
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Robby ROUDON <roudonro@msu.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+trait SecretTrait
+{
+    /**
+     * Load a secret value from the specified configuration and key.
+     * Will look for a _file-suffixed version of the key first,
+     * and load the data from a separate file if configured to do so.
+     *
+     * @param Config|array|null $config The config to read from
+     * @param string            $key    The key to retrieve
+     *
+     * @return string|null
+     */
+    protected function getSecretFromConfigOrSecretFile(Config|array|null $config, string $key): ?string
```

Comment:
I'd prefer naming this in a way that's agnostic to the source of the secret. That would make it possible to add support for other secret storage methods in future without changing the calls to this method. Could we use a simple name like `getSecret`?

Reply:
While I agree that the current name (which was my suggestion, so I'll take the blame for it) was too specific, I think `getSecret` may be a little too vague (since we're using a trait and it's best to avoid potential collisions). Maybe `getConfiguredSecret` or `getSecretFromConfig` would be a good compromise -- since we're passing configuration to the method, I think either name is accurate even if we're using special methods to process the configuration.

Reply:
I changed it to `getSecretFromConfig`
For this converation and all other I leave you close them if you consider them resolved

Reply:
`getSecretFromConfig` works for me!